src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +941,945 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+    ObjectGuid _playerGUID;
+    ObjectGuid _yorickGUID;
+    ObjectGuid _crowleyGUID;
+    ObjectGuid _bloodfangGUID;
+};
+
+enum DariusCrowleyExsanguinate
+{
+    WAYPOINT_ON_CROWLEY_DESPAWN              = 2
+};
+
+// Lord Darius Crowley - 44883
+struct npc_silverpine_lord_darius_crowley_exsanguinate : public ScriptedAI
+{
+    npc_silverpine_lord_darius_crowley_exsanguinate(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+            _playerGUID = player->GetGUID();
+
+        FindAllGuid();
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_CROWLEY_EXIT && waypointId == WAYPOINT_ON_CROWLEY_DESPAWN)
+            me->DespawnOrUnsummon();
+    }
+
+    void FindAllGuid()
+    {
+        if (!_bloodfangGUID)
+        {
+            if (Creature* bloodfang = me->FindNearestCreature(NPC_PACKLEADER_IVAR_BLOODFANG, 100.0f))
+                _bloodfangGUID = bloodfang->GetGUID();
+        }
+
+        if (!_armoireGUID)
+        {
+            if (Creature* armoire = me->FindNearestCreature(NPC_ARMOIRE_SUMMONED, 100.0f))
+            {
+                _armoireGUID = armoire->GetGUID();
+
+                if (armoire->IsAIEnabled())
+                    armoire->GetAI()->SetGUID(me->GetGUID(), me->GetEntry());
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    ObjectGuid _armoireGUID;
+    ObjectGuid _bloodfangGUID;
+};
+
+enum IvarBloodfangExsanguinate
+{
+    WAYPOINT_ON_BLOODFANG_DESPAWN            = 3
+};
+
+// Packleader Ivar Bloodfang - 44884
+struct npc_silverpine_packleader_ivar_bloodfang_exsanguinate : public ScriptedAI
+{
+    npc_silverpine_packleader_ivar_bloodfang_exsanguinate(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+            _playerGUID = player->GetGUID();
+
+        if (Creature* armoire = me->FindNearestCreature(NPC_ARMOIRE_SUMMONED, 30.0f))
+        {
+            _armoireGUID = armoire->GetGUID();
+
+            if (armoire->IsAIEnabled())
+                armoire->GetAI()->SetGUID(me->GetGUID(), me->GetEntry());
+        }
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_BLOODFANG_EXIT && waypointId == WAYPOINT_ON_BLOODFANG_DESPAWN)
+            me->DespawnOrUnsummon();
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    ObjectGuid _armoireGUID;
+};
+
+enum EjectPassenger1
+{
+    SEAT_BLOODFANG                          = 0
+};
+
+// Eject Passenger 1 - 80743
+class spell_silverpine_eject_passenger_1 : public SpellScript
```

Comment:
Maybe one of scripts in spell_gen can be used, there's one generic

Reply:
There's spell_gen_eject_passenger which calculates it off the value by substracting -1, but this one cannot be used since 80743 has value 0 and spell_gen_eject_passenger_with_seatId targets any possible passenger and it should only be passenger on seat 0.

However, I do think it would be wiser to move this one as a generic spell to only target seat 0 instead.